contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2088,6 +2089,292 @@ numeric_radians(PG_FUNCTION_ARGS)
+                    initial_state[current_part] = 2;
+                }
+
+                start_positions[current_part] = &object_name_str[i + 1];
+                i += consumed;
+                continue;
+            }
+            else if (c == '.')
+            {
+                // do not update the value of end_positions[current_part] if there is already a value in end_postions[current_part] & previous character is " or ].
+                if ( !((end_positions[current_part] != NULL) && (object_name_str[i - 1] == '"')) && !((end_positions[current_part] != NULL) && (object_name_str[i - 1] == ']')) )
+                {
+                    end_positions[current_part] = &object_name_str[i - 1];
+                }
+
+                current_part++;
+                if (current_part > 3)
+                {
+                    PG_RETURN_NULL();
+                }
+
+                start_positions[current_part] = &object_name_str[i + 1];
+                total_chars = 0;
+                total_length = 0;
+            }
+        }
+        else if (state == 1)
+        {
+            if (c == '"')
+            {
+                // is there a next character and is it double quotes?
+                if (i + consumed < len && object_name_str[i + consumed] == '"')
+                {
+                    i += consumed;
+                }
+                else
+                {
+                    state = 0;
+                    end_positions[current_part] = &object_name_str[i - 1];
+                    if (i + 1 < len && object_name_str[i + 1] != '.')
+                    {
+                        PG_RETURN_NULL();
+                    }
+                    i += consumed;
+                    continue;
+                }
+            }
+        }
+        else if (state == 2)
+        {
+            if (c == ']')
+            {
+                // is there a next character and if it is there, is it closing brace?
+                if (i + consumed < len && object_name_str[i + consumed] == ']')
+                {
+                    i += consumed;
+                }
+                else
+                {
+                    state = 0;
+                    end_positions[current_part] = &object_name_str[i - 1];
+                    if (i + 1 < len && object_name_str[i + 1] != '.')
+                    {
+                        PG_RETURN_NULL();
+                    }
+                    i += consumed;
+                    continue;
+                }
+            }
+        }
+
+        // just add the total characters and total length
+        if (state > 0 || (state == 0 && c != '.'))
+        {
+            if (code <= 0xFFFF)
+                total_chars += 1;
+            else
+                total_chars += 2;
+            total_length += (code <= 0xFFFF) ? 2 : 4;
+        }
+        i += consumed;
+    }
+
+    if (state != 0)
+    {
+        PG_RETURN_NULL();
+    }
+
+    if (total_chars > 128 || total_length > 256)
```

Comment:
How does the parsing handle cases where there is an invalid UTF8 character? i.e. a new line character